src/sage/rings/species.py

```diff
@@ -0,0 +1,1027 @@
+from itertools import accumulate, chain, combinations
+from math import prod
+from sage.arith.misc import binomial
+from sage.categories.cartesian_product import cartesian_product
+from sage.categories.graded_algebras_with_basis import GradedAlgebrasWithBasis
+from sage.categories.infinite_enumerated_sets import InfiniteEnumeratedSets
+from sage.categories.monoids import Monoids
+from sage.categories.sets_with_grading import SetsWithGrading
+from sage.combinat.free_module import CombinatorialFreeModule
+from sage.combinat.integer_vector import IntegerVectors
+from sage.combinat.partition import Partitions
+from sage.combinat.permutation import Permutations
+from sage.groups.perm_gps.constructor import PermutationGroupElement
+from sage.groups.perm_gps.permgroup import PermutationGroup, PermutationGroup_generic
+from sage.groups.perm_gps.permgroup_named import SymmetricGroup
+from sage.libs.gap.libgap import libgap
+from sage.misc.cachefunc import cached_method
+from sage.misc.lazy_attribute import lazy_attribute
+from sage.monoids.indexed_free_monoid import IndexedFreeAbelianMonoid, IndexedFreeAbelianMonoidElement, IndexedMonoid
+from sage.rings.integer_ring import ZZ
+from sage.structure.element import Element, parent
+from sage.structure.parent import Parent
+from sage.structure.unique_representation import UniqueRepresentation
+from sage.sets.finite_enumerated_set import FiniteEnumeratedSet
+
+GAP_FAIL = libgap.eval('fail')
+
+
+def _is_conjugate(G, H1, H2):
+    r"""
+    Test if ``H1`` and ``H2`` are conjugate subgroups in ``G``.
+
+    EXAMPLES::
+
+        sage: G = SymmetricGroup(3)
+        sage: H1 = PermutationGroup([(1,2)])
+        sage: H2 = PermutationGroup([(2,3)])
+        sage: from sage.rings.species import _is_conjugate
+        sage: _is_conjugate(G, H1, H2)
+        True
+    """
+    return GAP_FAIL != libgap.RepresentativeAction(G, H1, H2)
```

Comment:
I think, in the end we should use `IsConjugate` here, and actually, we should make this a method of `PermutationGroup_generic`.